modules/app.js

```diff
@@ -0,0 +1,144 @@
+  item.completed ? 'checked' : ''
+} />
+    <p>${item.description}</p>
+  </label>
+
+  <button class="detail-btn" data-index="${item.index}">
+    <i class="fa-solid fa-ellipsis-vertical" id="vertical"></i>
+  </button>
+
+  <button class="delete-btn hide" data-index="${item.index}">
+    <i class="fa-regular fa-trash-can" id="trash"></i>
+  </button>
+</div>
+<hr />
+    `,
+  );
+  lists.innerHTML = displayData.join(' ');
+
+  // toggle checked and unchecked on the list
+  toggleCheckbox(todoArray);
+
+  // add event listener to detail buttons
+  const detailbtn = document.querySelectorAll('.detail-btn');
+  detailbtn.forEach((btn) => {
+    btn.addEventListener('click', (e) => {
+      e.preventDefault();
+      const index = parseInt(e.currentTarget.dataset.index, 10);
+      const deletebtn = document.querySelector(
+        `.delete-btn[data-index="${index}"]`,
+      );
+      deletebtn.classList.remove('hide');
+      btn.classList.add('hide');
+
+      // make the paragraph editable
+      const p = e.currentTarget.parentNode.querySelector('p');
+      p.contentEditable = true;
+      p.focus();
+    });
+  });
+
+  // add event listener to delete buttons
+  const deletebtn = document.querySelectorAll('.delete-btn');
+  deletebtn.forEach((btn) => {
+    btn.addEventListener('click', (e) => {
+      const index = parseInt(e.currentTarget.dataset.index, 10);
+      todoArray.splice(index - 1, 1);
+      for (let i = index - 1; i < todoArray.length; i += 1) {
+        todoArray[i].index = i + 1;
+      }
+      addData();
+      populateList();
+    });
+  });
+
+  // Delete all
+  const clearBtn = document.getElementById('clear-btn');
+  clearBtn.addEventListener('click', (e) => {
+    // eslint-disable-next-line no-unused-vars
+    const index = parseInt(e.currentTarget.dataset.index, 10);
+    todoArray = todoArray.filter((item) => !item.completed);
+    todoArray.forEach((item, index) => {
+      item.index = index + 1;
+    });
+    addData();
+    populateList();
+  });
+
+  // add event listener to paragraphs for editing
+  const paragraphs = document.querySelectorAll('.list-item p');
+  paragraphs.forEach((p, i) => {
+    p.addEventListener('keydown', (e) => {
+      if (e.keyCode === 13) {
+        e.preventDefault();
+        p.blur();
+      }
+    });
+    p.addEventListener('blur', () => {
+      todoArray[i].description = p.textContent.trim;
+      addData();
+    });
+  });
+};
+
+// add list when form is submitted
+const addlist = () => {
+  submit.addEventListener('click', (e) => {
+    e.preventDefault();
+    const inputData = input.value.trim();
+    if (inputData !== '') {
+      addItem(inputData);
+      input.value = '';
+      populateList();
+    }
+  });
+  input.addEventListener('keydown', (e) => {
+    if (e.keyCode === 13) {
+      e.preventDefault();
+      submit.click();
+    }
+  });
+};
+
+export {
+  getData, populateList, addlist, addData,
+};
```

Comment:
- [ ] Kindly consider separating all your files in different files to follow the best practices of [KISS](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/clean-code/lesson_DRY_KISS_YAGNI.md)

Reply:
👌